Sources/StreamChat/WebSocketClient/Events/PollsEvents.swift

```diff
@@ -0,0 +1,233 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+struct PollClosedEventDTO: EventDTO {
+    var poll: PollDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, poll: PollDTO? = nil) {
+        self.payload = payload
+        self.poll = poll
+    }
+}
+
+struct PollCreatedEventDTO: EventDTO {
+    var poll: PollDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, poll: PollDTO? = nil) {
+        self.payload = payload
+        self.poll = poll
+    }
+}
+
+struct PollDeletedEventDTO: EventDTO {
+    var poll: PollDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, poll: PollDTO? = nil) {
+        self.payload = payload
+        self.poll = poll
+    }
+}
+
+struct PollUpdatedEventDTO: EventDTO {
+    var poll: PollDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, poll: PollDTO? = nil) {
+        self.payload = payload
+        self.poll = poll
+    }
+}
+
+struct PollVoteCastedEventDTO: EventDTO {
+    var vote: PollVoteDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, vote: PollVoteDTO? = nil) {
+        self.payload = payload
+        self.vote = vote
+    }
+}
+
+struct PollVoteChangedEventDTO: EventDTO {
+    var vote: PollVoteDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, vote: PollVoteDTO? = nil) {
+        self.payload = payload
+        self.vote = vote
+    }
+}
+
+struct PollVoteRemovedEventDTO: EventDTO {
+    var vote: PollVoteDTO?
+    var payload: EventPayload
+
+    init(payload: EventPayload, vote: PollVoteDTO? = nil) {
+        self.payload = payload
+        self.vote = vote
+    }
+}
+
+struct PollDTO: Decodable {
+    var allowAnswers: Bool
+    var allowUserSuggestedOptions: Bool
+    var answersCount: Int
+    var createdAt: Date
+    var createdById: String
+    var description: String
+    var enforceUniqueVote: Bool
+    var id: String
+    var name: String
+    var updatedAt: Date
+    var voteCount: Int
+    var latestAnswers: [PollVoteDTO?]
+    var options: [PollOptionDTO?]
+    var ownVotes: [PollVoteDTO?]
+    var custom: [String: RawJSON]?
+    var latestVotesByOption: [String: [PollVoteDTO?]]
+    var voteCountsByOption: [String: Int]
+    var isClosed: Bool?
+    var maxVotesAllowed: Int?
+    var votingVisibility: String?
+    var createdBy: UserPayload?
```

Comment:
This should be Payload. DTO's (in our codebase) are used for CoreData models
```suggestion
struct PollPayload: Decodable {
    var allowAnswers: Bool
    var allowUserSuggestedOptions: Bool
    var answersCount: Int
    var createdAt: Date
    var createdById: String
    var description: String
    var enforceUniqueVote: Bool
    var id: String
    var name: String
    var updatedAt: Date
    var voteCount: Int
    var latestAnswers: [PollVoteDTO?]
    var options: [PollOptionDTO?]
    var ownVotes: [PollVoteDTO?]
    var custom: [String: RawJSON]?
    var latestVotesByOption: [String: [PollVoteDTO?]]
    var voteCountsByOption: [String: Int]
    var isClosed: Bool?
    var maxVotesAllowed: Int?
    var votingVisibility: String?
    var createdBy: UserPayload?
```